src/dogecoin-fees.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2021 The Dogecoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <boost/random/uniform_int.hpp>
+
+
+#include "policy/policy.h"
+#include "arith_uint256.h"
+#include "dogecoin.h"
+#include "txmempool.h"
+#include "util.h"
+#include "validation.h"
+#include "dogecoin-fees.h"
+#include "amount.h"
+#include <cmath>
+
+//mlumin: concept:  min relay fee 0.0001 doge; 
+//mlumin: concept:  min fee configurable and defaulted at 0.01 DOGE later
+
+
+//mlumin 5/2021: walletfees, all attached to GetDogecoinWalletFeeRate which is just the newly exposed ::minWalletTxFee
+CAmount GetDogecoinWalletFee(size_t nBytes_)
+{
+    //mlumin: super simple fee calc for dogecoin
+    //CAmount nFee = ((int)ceil(nBytes_/1000.0f));
+    CAmount nFee=GetDogecoinWalletFeeRate().GetFee(nBytes_);
+}
+
+CAmount GetDogecoinWalletFeeWithDust(const CTransaction& tx, unsigned int nBytes)
+{
+    CFeeRate walletFeeRate=GetDogecoinWalletFeeRate();
+    //mlumin: get the staged or created tx from the send ui - not sure we need to lock mempool in this case.
+    {
+        LOCK(mempool.cs);//mlumin: do we need to lock the mempool when preparing to transmit a wallet tx? 
+        uint256 hash = tx.GetHash();
+        double dPriorityDelta = 0;
+        CAmount nFeeDelta = 0;
+        mempool.ApplyDeltas(hash, dPriorityDelta, nFeeDelta);
+        if (dPriorityDelta > 0 || nFeeDelta > 0)
+            return 0;
+    }
+
+    //mlumin: using walletfeerate here for accessibility and clarity, protest if you like pls
+    CAmount fee= walletFeeRate.GetFee(nBytes);
+    fee += GetDogecoinDustFee(tx.vout, walletFeeRate);
+
+    //mlumin 5/2021: No need to check here for maximum tx size ever possible. Just return the fee.
+    return fee;
+}
+
+//mlumin 5/2021: Establish a wallet rate of n koinu per kb.
+//mlumin: this is somewhat redundant to the globally exposed ::minWalletTxFee, but honestly I'd like to have both the rate and amount (with size) here
+CFeeRate GetDogecoinWalletFeeRate()
+{
+    //mlumin 5/2021: currently 1x COIN or 1 dogecoin or 100,000,000 koinu
+    return ::minWalletTxFeeRate;
+}
```

Comment:
Do these need an `#ifdef ENABLE_WALLET` wrapper?

Reply:
they do, yes. added to todo